compiler/qsc_formatter/src/main.rs

```diff
@@ -169,17 +180,72 @@ fn main() -> Result<(), String> {
             println!("\t{f}");
         }
         println!("Run the formatter with the `--write` option to correct formatting for the above files.{Reset}");
-        Err("Files are not formatted.".to_string())
-    } else if !file_walker.skipped_files.is_empty() {
+    } else {
         println!(
-            "{Skip}{} files are correctly formatted. {} files skipped.{Reset}",
-            file_walker.file_count,
-            file_walker.skipped_files.len()
+            "{Passing}{} files are correctly formatted.{Reset}",
+            file_walker.file_count
         );
-        Ok(())
-    } else {
-        println!("{Passing}All files are correctly formatted.{Reset}");
-        Ok(())
+    }
+    if are_skipped_files {
+        println!("{Skip}Skipped {} files:", file_walker.skipped_files.len());
+        for f in file_walker.skipped_files.iter() {
+            println!("\t{f}");
+        }
+        print!("{Reset}");
+    }
+
+    match (file_walker.is_write, are_changed_files, are_skipped_files) {
+        (true, true, true) => {
+            println!(
+                "{Skip}Updated {} files. {} files skipped.{Reset}",
+                file_walker.changed_files.len(),
+                file_walker.skipped_files.len()
+            );
+            Err(format!(
+                "Could not read/write from {} files",
+                file_walker.skipped_files.len()
+            ))
+        }
+        (true, true, false) => Ok(()),
```

Comment:
Love the detailed error handling, but I'd probably move all the success cases to the top to make it more immediately obvious what the success conditions are.

Reply:
done